pygmt/clib/session.py

```diff
@@ -1987,12 +1997,19 @@ def virtualfile_to_grid(
         ...     outgrid = None
         ...     with lib.virtualfile_out(kind="grid", fname=outgrid) as voutgrd:
         ...         lib.call_module("read", f"@earth_relief_01d_g {voutgrd} -Tg")
-        ...         result = lib.virtualfile_to_grid(vfname=voutgrd, outgrid=outgrid)
+        ...         result = lib.virtualfile_to_raster(vfname=voutgrd, outgrid=outgrid)
         ...         assert isinstance(result, xr.DataArray)
         """
         if outgrid is not None:
             return None
-        return self.read_virtualfile(vfname, kind="grid").contents.to_dataarray()
+        if kind is None:  # Inquire the data family from the virtualfile
+            family = self.inquire_virtualfile(vfname)
+            kind = {
+                self["GMT_IS_GRID"]: "grid",
+                self["GMT_IS_IMAGE"]: "image",
+                self["GMT_IS_CUBE"]: "cube",
+            }[family]
```

Comment:
Can we use a match statement here like this? (untested)
```suggestion
            match family
                case self["GMT_IS_GRID"]:
                    kind = "grid"
                case self["GMT_IS_IMAGE"]:
                    kind = "image"
                case self["GMT_IS_CUBE"]:
                    kind = "cube"
```

Reply:
Dict-lookup is faster than match statement (https://stackoverflow.com/questions/68476576/python-match-case-switch-performance).